src/python/pants/backend/python/goals/package_dists.py

```diff
@@ -9,27 +9,19 @@
 from pants.backend.python.util_rules.package_dists import create_dist_build_request
 from pants.backend.python.util_rules.package_dists import rules as package_dists_rules
 from pants.core.goals.package import BuiltPackage, BuiltPackageArtifact, PackageFieldSet
-from pants.engine.internals.native_engine import Address
 from pants.engine.intrinsics import digest_to_snapshot
 from pants.engine.rules import collect_rules, implicitly, rule
 from pants.engine.unions import UnionMembership, UnionRule
 
 
-@rule
-async def dist_target_address(
```

Comment:
This was an unnecessary bit of cleverness that I introduced in a [previous PR](https://github.com/pantsbuild/pants/pull/22327), thinking it would be necessary for union call-by-name. Looks like it isn't, and so might as well revert it to prove that.